lib/ui/painting.dart

```diff
@@ -42,9 +42,19 @@ bool _offsetIsValid(Offset offset) {
 bool _matrix4IsValid(Float64List matrix4) {
   assert(matrix4 != null, 'Matrix4 argument was null.');
   assert(matrix4.length == 16, 'Matrix4 must have 16 entries.');
+  assert(matrix4.every((double value) => value.isFinite), 'Matrix4 entries must be finite.');
   return true;
```

Comment:
Wont this mean that in profile and release modes, the engine will still end up with invalid matrices?

Reply:
Yes, but as a general principle, we do a ton of validation in debug mode only via asserts.

Reply:
Yes, but we're calling `_throwIfMarix4IsInvalid` in `pushTransform`.

Reply:
This was already true - if someone decided to pass null for example,  or use `NaN` in a `Rect` or `Offset`.